x-pack/plugins/ml/server/shared_services/providers/job_service.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller } from 'kibana/server';
+import { LicenseCheck } from '../license_checks';
+import { jobServiceProvider } from '../../models/job_service';
+
+export interface JobServiceProvider {
+  jobServiceProvider(callAsCurrentUser: APICaller): ReturnType<typeof jobServiceProvider>;
+}
+
+export function getJobServiceProvider(isFullLicense: LicenseCheck): JobServiceProvider {
+  return {
+    jobServiceProvider(callAsCurrentUser: APICaller) {
+      isFullLicense();
```

Comment:
Does this function call have a side effect like a `throw` or does this need change to be part of an `if`, in other files I see `if (mlLicense.isFullLicense() === false) {`.

Reply:
yes, it will throw an error if it is not a full license.